### PR TITLE
Closes softlayer/sl-ember-components#550

### PR DESCRIPTION
**Pull request by juwara0**
Fix deprecation message in sl-date-range-picker by incorporating the sl-component-input-id mixin, updated component template, docs and tests.

### CODE REVIEW

**Comment by notmessenger** [comment]
Failing test:

```
not ok 251 PhantomJS 1.9 - Unit | Component | sl date range picker: label is accepted as a parameter
    ---
        actual: >
            ember8201-input
        expected: >
            ember8215
        message: >
            label is used for start date input
        Log: |
    ...
```

File: tests/unit/components/sl-date-range-picker-test.js [hunk 1]
```diff
@@ -15,6 +16,13 @@ moduleForComponent(
     }
 );
 
+test( 'Expected Mixins are present', function( assert ) {
+    assert.ok(
+        ComponentInputId.detect( this.subject() ),
+        'ComponentInputId Mixin is present'
```
Comment (notmessenger) [inline tests/unit/components/sl-date-range-picker-test.js]: If going to reference the mixin name, would use the actual Mixin name, not the import alias.

Comment (juwara0) [inline tests/unit/components/sl-date-range-picker-test.js]: I have updated the name to be the actual mixin name

**Comment by juwara0** [comment]
On response to:
Failing test:

not ok 251 PhantomJS 1.9 - Unit | Component | sl date range picker: label is accepted as a parameter
    ---
        actual: >
            ember8201-input
        expected: >
            ember8215
        message: >
            label is used for start date input
        Log: |
    ...


This is expected because the date-range-picker has been updated to use the updates in https://github.com/softlayer/sl-ember-components/pull/806 but this is not present until that PR is merged in.

**Pull request merged**